### DISCUSSION TITLE
Why number field doesn't allow null?

### DISCUSSION THREAD

**Discussion by speryt** [pocketbase/pocketbase#4847] [Q&A] [answered]
What is the reasoning behind forcing empty `number` type field to be 0 instead of null? Or there is some workaround for that?
I have cases where 0 is correct value and I also have cases where I don't have number value. In later case I'd prefer to have N/A value as it is the case for example in `text` type fields.
I have seen issue #1011 but it is lacking a bit of explanation - why.

**Comment by ganigeorgiev** [comment, marked as answer]
With exception of the `json` field type, all other field types are non-nullable and uses zero-value by default (_the first value in the [Overview table](https://pocketbase.io/docs/collections/#overview)_).

About why - because it is easier to work with non-nullable types. For more details you can also check https://github.com/pocketbase/pocketbase/issues/122.

**Reply by MichaelThomas0721** [reply]
Would you be able to elaborate on why it is easier to work with non-nullable types? I am currently working with a dataset that contains numbers in the positives, negatives and 0, on top of that, it is still possible for the number to not exist which would invoke behavior different then if it was 0. 
What is the recommended alternative to null values? My current idea is to use a string instead of a number or have an additional column that indicates if it is null. Both solutions feel a little clunky in my opinion.
I would appreciate some insight from someone more experienced then me!

**Reply by ganigeorgiev** [reply]
@MichaelThomas0721 Please read the comments in the above linked issue https://github.com/pocketbase/pocketbase/issues/122

**Reply by MichaelThomas0721** [reply]
I guess I'll go with your suggestion of using a boolean.

I appreciate your time and effort on this project!

**Reply by lukepighetti** [reply]
would be nice to have an option for default value as -1 instead of 0
